allennlp/models/coreference_resolution/coref.py

```diff
@@ -472,15 +525,185 @@ def _generate_valid_antecedents(
         # In our matrix of indices, the upper triangular part will be negative
         # because the offsets will be > the target indices. We want to mask these,
         # because these are exactly the indices which we don't want to predict, per span.
-        # We're generating a logspace mask here because we will eventually create a
-        # distribution over these indices, so we need the 0 elements of the mask to be -inf
-        # in order to not mess up the normalisation of the distribution.
         # Shape: (1, num_spans_to_keep, max_antecedents)
-        valid_antecedent_log_mask = (raw_antecedent_indices >= 0).float().unsqueeze(0).log()
+        valid_antecedent_mask = (raw_antecedent_indices >= 0).unsqueeze(0)
 
         # Shape: (num_spans_to_keep, max_antecedents)
         valid_antecedent_indices = F.relu(raw_antecedent_indices.float()).long()
-        return valid_antecedent_indices, valid_antecedent_offsets, valid_antecedent_log_mask
+        return valid_antecedent_indices, valid_antecedent_offsets, valid_antecedent_mask
+
+    def _distance_pruning(
+        self,
+        top_span_embeddings: torch.FloatTensor,
+        top_span_mention_scores: torch.FloatTensor,
+        max_antecedents: int,
+    ) -> Tuple[torch.FloatTensor, torch.BoolTensor, torch.LongTensor, torch.LongTensor]:
+        """
+        Generates antecedents for each span and prunes down to `max_antecedents`.
```

Comment:
Can you add a more comprehensive description of the pruning methods here and below?